seerep-hdf5/seerep-hdf5-pb/test/pb-write-load-test.cpp

```diff
@@ -0,0 +1,327 @@
+#include <gtest/gtest.h>
+#include <seerep-core/core.h>
+#include <seerep-hdf5-pb/hdf5-pb-image.h>
+
+#include <H5File.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <ctime>
+#include <optional>
+#include <string>
+#include <vector>
+
+/*
+This test serves more as an integration test rather than a unit test. We want
+to make sure that a Protobuf image message is the same after saving and
+reading it from a hdf5 file. Because the underlying components like the
+seerep-hdf5-core and seerep-hdf5-pb are not (fully) tested, we don't use mocks and
+propagate the data through all components. This should be changed in the future
+to improve the quality of the tests.
+*/
+
+void createHeader(const std::string projectUUID, const std::string messageUUID, seerep::Header* header)
+{
+  /**
+   * @brief given a pointer to an image header, sets sub field values.
+   * @param[in] projectUUID the UUID of the project
+   * @param[in] messageUUId the UUID of the message
+   * @param[in,out] header the header which will hold the fields
+   **/
+  header->set_seq(5);
+  header->set_frame_id("arbitrary_id");
+  header->mutable_stamp()->set_seconds(5);
+  header->mutable_stamp()->set_nanos(5);
+  header->set_uuid_project(projectUUID);
+  header->set_uuid_msgs(messageUUID);
+}
+
+/**
+ * @brief sets values in a pointer to a point, given x and y co ordinate values
+ * @param[in] x x co ordinate
+ * @param[out] y y co ordinate
+ * @param[in,out] point2D pointer to a 2D point object
+ * */
+void createPoint(const double x, const double y, seerep::Point2D* point2D)
+{
+  point2D->set_x(x);
+  point2D->set_y(y);
+}
+
+/**
+ * @brief creates a grid of image data given height and width of an image
+ * @param[in] imageHeight the height of the image
+ * @param[in] imageWidth the width of the image
+ * @param[in,out] image address of the image object
+ * */
+void createImageData(const unsigned int imageHeight, const unsigned int imageWidth, seerep::Image& image)
+{
+  uint8_t data[imageHeight][imageWidth][3];
+  for (size_t i = 0; i < imageWidth; i++)
+  {
+    for (size_t j = 0; j < imageHeight; j++)
+    {
+      float x = float(i) / imageHeight;
+      float y = float(j) / imageHeight;
+      float z = float(j) / imageHeight;
+
+      uint8_t r = int((x * 255.0)) % 255;
+      uint8_t g = int((y * 255.0)) % 255;
+      uint8_t b = int((z * 255.0)) % 255;
+
+      data[j][i][0] = r;
+      data[j][i][1] = g;
+      data[j][i][2] = b;
+    }
+  }
+
+  image.set_data(data, sizeof(data));
+}
+
+/**
+ * @brief given a labelWithInstance set arbitrary label and uuid
+ * @param[in,out] labelWithInstance a pointer to a label with instance
+ * */
+void createLabelWithInstance(seerep::LabelWithInstance* labelWithInstance)
```

Comment:
Why do you use sometimes pointer and sometimes references as function arguments? Please use references or smart-pointers whenenver possible